### ISSUE TITLE
Please add a (n)oping option to ping all IP addresses (i.e. both, IPv4 and IPv6) for given hostnames

### ISSUE THREAD

**Issue by xtaran** [octo/liboping#52]
A very common noping use case for me is to check when a server comes up after reboot. For that I want to check both, IPv4 and IPv6 addresses of the same host, i.e. for the same hostname.

Currently I do this like this:

```noping $(host someserver.example.org | sed -e 's/^.* address //')```

It would be nice if noping would have an option to ping all IP addresses a hostname resolves, too, especially A and AAAA records. I suggest to use `-5` as it is between `-4` and `-6`, but `-a` or `-A` would be fine as well. Would then look like this:

```noping -5 someserver.example.org```

I guess oping itself (which I nearly never use) should get the same option, too.

**Comment by xtaran** [comment]
> I suggest to use `-5` as it is between `-4` and `-6`, but `-a` or `-A` would be fine as well.

Actually, in hexadecimal, `-a` or `-A` would be `-4` + `-6`, so in the meanwhile, I'd prefer and suggest `-a` or `-A` as it's even more fitting (and more geeky). ;-)

**Comment by cron2** [comment]
Since this is something I wanted as well, I have a patch for this :-) - octo seems to be quite busy so it did not get review yet.

I'll do a repo-clone-and-pullreq, so you can have a look and test...

**Comment by cron2** [comment]
Rebased to master, polished, split functionality into "multiping" and "canonify output hostnames" - we're using this internally since half a year, so it's generally working (on FreeBSD).

It would be cool if people could test this - then I'll do a PR so we can get this merged, hopefully :-)

https://github.com/cron2/liboping/

has the change, in the "multiping" branch.

**Comment by octo** [comment]
Sounds like a great feature! Could you create a PR, @cron2? That would make it easier to test and review.

Personally I would find `-a` for "all" quite intuitive for this behavior. We would also add `-A` for "automatic", i.e. the default behavior.

Best regards,
—octo

**Comment by cron2** [comment]
I'll rework what I have a bit and open a PR later today.

What I have now modifies the behaviour of ping_host_add() but neglects to update the manpage... I think I will add a ping_host_add_multiple() which takes a "max_hosts" argument - controlled by "-a" - and "ping_host_add()" will then become a wrapper for "ping_host_add_multiple( ... , 1)" with the existing calling convention.  And, man page :-)

**Comment by xtaran** [comment]
Hi,

yay on progress here! :-)

> Sounds like a great feature! Could you create a PR, @cron2? That would make it easier to test and review.

Thanks to @cron2 for providing an implementation.

> Personally I would find `-a` for "all" quite intuitive for this behavior.

Fine for me. :-)

> We would also add `-A` for "automatic", i.e. the default behavior.

Why would this be needed? Will the default be changed?

**Comment by xtaran** [comment]
Since this still didn't make it into liboping proper and hence won't be part of the upcoming Debian 11 Bullseye Stable release, I wrote my own wrapper around `noping` as I just once again ran into a situation where had to resolve dual-stack IP addresses manually. My wrapper implementation can be found at https://github.com/xtaran/nopinga/.

**Comment by barak** [comment]
Sorry about that! Would have been happy to cherry-pick that into the library if someone had poked me.
```
$ echo -{4,6}\ foo
-4 foo -6 foo
```

**Comment by xtaran** [comment]
> Sorry about that! Would have been happy to cherry-pick that into the library if someone had poked me.

Nah, don't worry. I actually would like to see this in liboping proper, not just cherry-picked in Debian. (I actually also have to use other Linux distributions...)
